course/src/main/java/org/openedx/course/presentation/container/CourseContainerAdapter.kt

```diff
@@ -2,16 +2,30 @@ package org.openedx.course.presentation.container
 
 import androidx.fragment.app.Fragment
 import androidx.viewpager2.adapter.FragmentStateAdapter
+import org.openedx.course.R
 
 class CourseContainerAdapter(fragment: Fragment) : FragmentStateAdapter(fragment) {
 
-    private val fragments = ArrayList<Fragment>()
+    private val fragments = HashMap<CourseContainerTab, Fragment>()
 
     override fun getItemCount(): Int = fragments.size
 
-    override fun createFragment(position: Int): Fragment = fragments[position]
+    override fun createFragment(position: Int): Fragment {
+        val tab = CourseContainerTab.values().find { it.ordinal == position }
+        return fragments[tab] ?: throw IllegalStateException("Fragment not found for tab $tab")
+    }
 
-    fun addFragment(fragment: Fragment) {
-        fragments.add(fragment)
+    fun addFragment(tab: CourseContainerTab, fragment: Fragment) {
+        fragments[tab] = fragment
     }
-}
\ No newline at end of file
+
+    fun getFragment(tab: CourseContainerTab): Fragment? = fragments[tab]
+}
+
+enum class CourseContainerTab(val itemId: Int, val titleResId: Int) {
+    OUTLINE(itemId = R.id.outline, titleResId = R.string.course_navigation_course),
```

Comment:
A better name would have been `Course` or `Home` but the file name is Outline.😅

Reply:
Updates with `course`